src/simulation/thermal/sensorThermal/sensorThermal.cpp

```diff
@@ -0,0 +1,233 @@
+        bskLogger.bskLog(BSK_ERROR, "sensorThermal.stateInMsg was not linked.");
+    }
+
+    this->T = this->T_0;
+
+    return;
+}
+
+bool SensorThermal::readMessages()
+{
+    //! - Zero ephemeris information
+    this->sunData = sunInMsg.zeroMsgPayload;
+    this->stateCurrent = stateInMsg.zeroMsgPayload;
+
+    //! - If we have a valid sun ID, read Sun ephemeris message
+    if(this->sunInMsg.isLinked())
+    {
+        this->sunData = this->sunInMsg();
+    }
+    //! - If we have a valid state ID, read vehicle state ephemeris message
+    if(this->stateInMsg.isLinked())
+    {
+        this->stateCurrent = this->stateInMsg();
+    }
+    //! - Read in optional sun eclipse input message 
+    if(this->sunEclipseInMsg.isLinked()) {
+        EclipseMsgPayload sunVisibilityFactor;          // sun visiblity input message
+        sunVisibilityFactor = this->sunEclipseInMsg();
+        this->shadowFactor = sunVisibilityFactor.shadowFactor;
+    }
+
+    DeviceStatusMsgPayload statusMsg;
+
+    //! - read in the device status msg
+    if(this->sensorStatusInMsg.isLinked())
+    {
+        statusMsg = this->sensorStatusInMsg();
+        this->sensorStatus = statusMsg.deviceStatus;
+    } else {
+        this->sensorStatus = 1;
+    }
+
+    return(true);
+}
+
+/*! Provides logic for running the read / compute / write operation that is the module's function.
+ @param CurrentSimNanos The current simulation time in nanoseconds
+ */
+void SensorThermal::UpdateState(uint64_t CurrentSimNanos)
+{
+
+    //! - Only update the thermal status if we were able to read in messages.
+    if(this->readMessages())
+    {
+        this->evaluateThermalModel(CurrentSimNanos*NANO2SEC);
+    } else {
+        /* if the read was not successful then zero the output message */
+        this->temperatureMsgBuffer = this->temperatureOutMsg.zeroMsgPayload;
+    }
+
+    this->writeMessages(CurrentSimNanos);
+
+    return;
+}
+
+/*! This method writes out a message.
+ @return void
+ */
+void SensorThermal::writeMessages(uint64_t CurrentClock)
+{
+    //! - write temperature output message
+    this->temperatureOutMsg.write(&this->temperatureMsgBuffer, this->moduleID, CurrentClock);
+
+    return;
+}
+
+/*!
+ *  This method allows for the sensor-specific parameters to be set
+ * @param nHat_B: normal direction of the sensor in the spacecraft body frame
+ * @param sensorArea: area of the sensor (units m^2), assumed to be a flat plate
+ * @param sensorAbsorptivity: coefficient of the sensor absorptivity (unitless, between 0 and 1)
+ * @param sensorEmissivity: coefficient of the sensor emissivity (unitless, between 0 and 1)
+ * @param sensorMass: mass of the sensor (kg)
+ * @param sensorSpecificHeat: specific heat of the material of the sensor (units J/kg/C)
+ */
+void SensorThermal::setSensorParameters(Eigen::Vector3d nHat_B, double sensorArea, double sensorAbsorptivity, double sensorEmissivity, double sensorMass, double sensorSpecificHeat)
```

Comment:
:large_orange_diamond: As discussed let's add some accessor methods to this module's api.

Reply:
Why do we even have this method?  All these variables are public variables and can be set directly via `module.sensorArea`, etc.?

Reply:
There are a subset of the variables that are required. Adam and I discussed that these required variables should move into the constructor, so they are by definition required.

Reply:
I agree with that part.  I see the required parameters being set to non-sense values, and then in Reset() we check the user has set required parameters, and other parameters are set to sensible values.  I guess I don't understand the issue being raised.  My comment is that I don't see the use of the `setSensorParameters` method.  It doesn't make my python code easier to read or use.

Reply:
What I'm hearing is to A.) Add the required parameters to the constructor, and B.) have the optional parameters set with `model.Parameter = value`?

Reply:
In many realms of software engineering, the idea of a required parameter, is one where without the parameter, you cannot proceed/have an incomplete definition, which should raise an error/exception alerting the user to this fact. Ideally, the user wouldn't be able to use the element/capability without having provided the required parameters. This helps provide a conceptually clearer experience for the user. One where the API communicates it's intent through mechanism as well as documentation. The example of `model.parameterRequire = value` does nothing to ensure the user is correctly configuring the element. Just my 2 cents 🙂

Reply:
What we did with BSK2 is all the models now check, if possible, in the `Reset()` method if required parameters are properly set.  This can check if the value is set to something, as well as that this something is with a correct range of values.  For example, if mass must be defined as a positive value, then we check in reset if `m <= 0` and if yes throw BSK_ERROR to let the user know.

I would be in favor of not defining default module here.  We define all required parameters to values that we can check in Reset() to ensure they have been set.  This is the behavior we use in many other modules.

Reply:
Removed setter function. All parameters now set directly. Updated documentation, scenario, and unit tests accordingly.